src/main/java/org/folio/circulation/services/ActualCostRecordService.java

```diff
@@ -0,0 +1,78 @@
+package org.folio.circulation.services;
+
+import java.time.ZonedDateTime;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.folio.circulation.domain.ActualCostRecord;
+import org.folio.circulation.domain.FeeFine;
+import org.folio.circulation.domain.FeeFineOwner;
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.ItemLossType;
+import org.folio.circulation.infrastructure.storage.ActualCostRecordRepository;
+import org.folio.circulation.support.results.Result;
+import org.folio.circulation.support.utils.ClockUtil;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.domain.FeeFine.LOST_ITEM_ACTUAL_COST_FEE_TYPE;
+import static org.folio.circulation.services.LostItemFeeChargingService.ReferenceDataContext;
+import static org.folio.circulation.support.results.Result.succeeded;
+import static org.folio.circulation.support.results.ResultBinding.mapResult;
+
+public class ActualCostRecordService {
+  private final ActualCostRecordRepository actualCostRecordRepository;
+
+  public ActualCostRecordService(ActualCostRecordRepository actualCostRecordRepository) {
+    this.actualCostRecordRepository = actualCostRecordRepository;
+  }
+
+  public CompletableFuture<Result<ReferenceDataContext>> createActualCostRecordIfNecessary(
+    ReferenceDataContext referenceDataContext) {
+
+    return createActualCostRecordIfNecessary(referenceDataContext.getLoan(),
+      referenceDataContext.getFeeFineOwner(),
+      ItemLossType.DECLARED_LOST, ClockUtil.getZonedDateTime(),
```

Comment:
Formatting of the parameters - we usually only wrap them when they start exceeding the limit of 100 chars

Reply:
the params are formatted, thanks.